tests/post_training/pipelines/image_classification_timm.py

```diff
@@ -112,29 +120,35 @@ def transform_fn(data_item):
 
     def prepare_calibration_dataset(self):
         dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
-        loader = torch.utils.data.DataLoader(dataset, batch_size=1, num_workers=2, shuffle=False)
+        loader = torch.utils.data.DataLoader(dataset, batch_size=self.batch_size, num_workers=2, shuffle=False)
 
         self.calibration_dataset = nncf.Dataset(loader, self.get_transform_calibration_fn())
 
     def _validate(self):
         val_dataset = datasets.ImageFolder(root=self.data_dir / "imagenet" / "val", transform=self.transform)
-        val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=1, num_workers=2, shuffle=False)
-
-        dataset_size = len(val_loader)
-
-        predictions = [0] * dataset_size
-        references = [-1] * dataset_size
+        val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=self.batch_size, num_workers=2, shuffle=False)
```

Comment:
What's the reason to validate with a batch?
Did you compare how it affect of time validation?
batch_size from description associated only with calibration_dataset.